app/models/activity.rb

```diff
@@ -0,0 +1,70 @@
+class Activity < ActiveRecord::Base
+  before_create :generate_event_key
+
+  belongs_to :object, polymorphic: true
+  belongs_to :actor, inverse_of: :activities, class_name: User.name
+  has_many :course_notifications, inverse_of: :activity, dependent: :destroy,
+                                  class_name: Course::Notification
+  has_many :user_notifications, inverse_of: :activity, dependent: :destroy
+
+  USER_NOTIFICATION_TYPES = [:popup, :email]
+  COURSE_NOTIFICATION_TYPES = [:feed, :email]
+
+  # @overload notify(to, types)
+  #   Create user notifications based on an activity.
+  #   @param [User] to the user who will receive notifications
+  #   @param [Array<Symbol>] types the formats of the notifications. It can be [:popup, :email]
+  # @overload notify(to, type)
+  #   Create user notifications based on an activity.
+  #   @param [User] to the user who will receive notifications
+  #   @param [Symbol] types the formats of the notifications. It can be :popup or :email
+  # @overload notify(to, types)
+  #   Create course notifications based on an activity.
+  #   @param [Course] to the course whose users will all receive notifications
+  #   @param [Array<Symbol>] types the formats of the notifications. It can be [:feed, :email]
+  # @overload notify(to, type)
+  #   Create course notifications based on an activity.
+  #   @param [Course] to the course whose users will all receive notifications
+  #   @param [Symbol] types the formats of the notifications. It can be :feed or :email
+  def notify(to, types: nil, type: nil)
+    types ||= [*type]
```

Comment:
What I did is that if a user input both types and type, I will only use types. Because I use `||=`